src/main/resources/db/migration/V202311131230__trusted_ca_mark.sql

```diff
@@ -0,0 +1,3 @@
+ALTER TABLE certificate ADD COLUMN trusted_ca BOOLEAN NULL;
+UPDATE certificate
+SET trusted_ca = CASE WHEN basic_constraints LIKE '%Subject Type=CA%' AND trusted_ca is NULL THEN false ELSE NULL END;
```

Comment:
It is not necessary to use CASE WHEN construct. Also expression `trusted_ca is NULL` will be always true since it is new column.

You just need to set it to false and add `WHERE` condition with `LIKE` as you have it.